### PR TITLE
Fix set artwork. Remove warning for RN 0.50. Fix set listener

### PR DESCRIPTION
**Pull request by kesha-antonov**
Hello! @tanguyantoine 

#### What's this PR do?

1) It cleans up code and set correct artwork when ```updatePlayback```/```setPlayback``` called
2) Remove warning for RN 0.50
3) Fix set listener of "RNMusicControlEvent". We should set it only if it not existed. Not on each call of "on"

### CODE REVIEW

File: ios/MusicControlManager.m [hunk 1]
```diff
@@ -231,36 +236,41 @@ - (void)sendEvent:(NSString*)event {
...
-             url = [artwork valueForKey: @"uri"];
-        }
-    }
+- (NSString*)getArtworkUrl:(NSString*)artwork {
+  NSString *artworkUrl = nil;
+
+  if (artwork) {
+      if ([artwork isKindOfClass:[NSString class]]) {
+           artworkUrl = artwork;
+      } else if ([[artwork valueForKey: @"uri"] isKindOfClass:[NSString class]]) {
```
Comment (tanguyantoine) [inline ios/MusicControlManager.m]: If you expect artwork to be a `NSString` type (from parameter type) then we won't ever pass in this condition. Right ?

**Comment by tanguyantoine** [comment]
Thank you for the PR. May I ask you to split it in multiple PR?

File: ios/MusicControlManager.m (hunk 1 above)
Comment (kesha-antonov) [reply ios/MusicControlManager.m]: I'm not obj-c expert. But it works

**Comment by kesha-antonov** [comment]
Hello!
Any trouble understanding code?

**Comment by tanguyantoine** [comment]
nope, but now there is conflict 😁

**Comment by tanguyantoine** [comment]
@DanNi0130 may I ask you to review?

**Comment by DanNi0130** [comment]
I'll take a look at this tomorrow morning!

**Comment by kesha-antonov** [comment]
Resolved

File: index.ios.js [hunk 2]
```diff
@@ -30,22 +38,20 @@ var MusicControl = {
   RATING_5_STARS: 0,
   RATING_PERCENTAGE: 0,
 
+  // NOTE: BACKWARDS COMPATIBILITY. USE "updatePlayback" INSTEAD.
   setPlayback: function (info) {
-    // Backwards compatibility. Use updatePlayback instead.
+    info = formatInfo(info);
     NativeMusicControl.updatePlayback(info);
```
Comment (tanguyantoine) [inline index.ios.js]: Could you remove variable creation like this

```js
NativeMusicControl.updatePlayback(formatInfo(info));
```

File: index.ios.js [hunk 3]
```diff
@@ -30,22 +38,20 @@ var MusicControl = {
...
   },
   updatePlayback: function(info) {
+    info = formatInfo(info);
     NativeMusicControl.updatePlayback(info);
```
Comment (tanguyantoine) [inline index.ios.js]: same

File: index.ios.js [hunk 4]
```diff
@@ -30,22 +38,20 @@ var MusicControl = {
...
     NativeMusicControl.enableBackgroundMode(enable)
   },
   setNowPlaying: function(info){
-    // Check if we have an ios asset from react style image require
-    if(info.artwork) {
-        info.artwork = resolveAssetSource(info.artwork) || info.artwork;
-    }
-
+    info = formatInfo(info);
     NativeMusicControl.setNowPlaying(info);
```
Comment (tanguyantoine) [inline index.ios.js]: same

**Comment by kesha-antonov** [comment]
@tanguyantoine I've updated what you asked.
Also I've noticed that there's ```index.js.android``` file which is almost the same. So I refactored ```index.js.ios/android``` into one ```index.js``` to prevent copy-paste code

**Comment by tanguyantoine** [comment]
Please see #79 for file merging. @DanNi0130 if it is okay for you, feel free to merge (without index merging, another PR is preferable)

**Comment by kesha-antonov** [comment]
Oops. Missed it. Alright guys. It's up to you to decide what to do next

**Comment by kesha-antonov** [comment]
I think it's better to merge my PR. Or you'll have to resolve conflicts 2 times

**Comment by DanNi0130** [comment]
@tanguyantoine, this looks fine to me. Should I merge this?

**Comment by tanguyantoine** [comment]
I would rather not merging this PR because of the index merge. If it was done in 2 PR then yes (easier to rollback if necessary)

Again, the fix is :+1: and thank to you. But this PR does well too much. 

If @Guichaguri agreed then let’s go.

**Comment by Guichaguri** [comment]
The index looks a bit messier, I agree that it should be moved to another PR, but I don't see any problems with this being merged

**Comment by kesha-antonov** [comment]
Come on guys. It's not a rocket science. Not much code changed

**Comment by tanguyantoine** [comment]
Rocket Science or not We prefer to have two different PRs. We sincerely appreciate your stuff but 2 PR make more sense

**Comment by kesha-antonov** [comment]
Ok. Sorry I do not have time now for this

**Comment by tanguyantoine** [comment]
I will try to backport you changes is I find the time.

**Comment by kesha-antonov** [comment]
Ok thanks!
I'll try to separate these to PR when I get back to this lib

**Comment by kesha-antonov** [comment]
I see this not getting merged. Closing

**Pull request closed without merging**